strategies.py

```diff
@@ -156,14 +156,10 @@ def play_move(self, c):
             self.searches_pi.append(self.root.children_as_pi(
                 self.root.position.n < self.temp_threshold))
         self.comments.append(self.root.describe())
-        try:
-            self.root = self.root.maybe_add_child(coords.to_flat(c))
-        except go.IllegalMove:
-            dbg("Illegal move")
-            if not self.two_player_mode:
-                self.searches_pi.pop()
-            self.comments.pop()
-            return False
+        self.root = self.root.maybe_add_child(coords.to_flat(c))
+        if not self.two_player_mode:
+            self.searches_pi.pop()
```

Comment:
These two pops should only be done if maybe_add_child throws, so I think you need to keep the try/catch block and reraise instead of returning false.